lib/drivers/medtronic600/medtronic600Driver.js

```diff
@@ -126,103 +144,63 @@ class BCNLMessage {
-      let timeoutFlag = false;
-      const abortTimer = setTimeout(() => {
-        debug('TIMEOUT');
-        timeoutFlag = true;
-        return reject(new TimeoutError('Timeout error.'));
-      }, readTimeout);
-
-      let message = Buffer.alloc(0);
-
-      async.doWhilst(
-        (callback) => {
-          hidDevice.receive((raw) => {
-            const packet = Buffer.from(new Uint8Array(raw));
-
-            // Only process if we get data
-            if (packet.length === 0) {
-              return callback(false);
-            }
-
-            const header = packet.slice(0, 3).toString('ascii');
-            const size = packet[3];
-
-            if (header !== BCNLMessage.MAGIC_HEADER) {
-              debug('Invalid packet from Contour device');
-              clearTimeout(abortTimer);
-              return callback(new InvalidMessageError('Unexpected USB packet header.'), null);
-            }
-
-            message = Buffer.concat([message, packet.slice(4)],
-              (message.length + packet.length) - 4);
-
-            // USB_BLOCKSIZE - 4, because we don't include the MAGIC_HEADER or the size byte
-            if (size < (BCNLMessage.USB_BLOCKSIZE - 4)) {
-              clearTimeout(abortTimer);
-              return callback(null, false); // We don't have to readMoreBytes
-            }
-            return callback(null, true); // We have to readMoreBytes
-          });
-        },
-        readMoreBytes => readMoreBytes, // Shall we doWhilst again?
-        (err) => {
-          if (err instanceof Error) {
-            throw err;
-          }
-          if (!timeoutFlag) {
-            debug('### READ USB DATA', message.toString('hex'));
-            resolve(message);
-          }
-        } // eslint-disable-line comma-dangle, for node
-      );
-    });
+  async readMessage(hidDevice, readTimeout = BCNLMessage.READ_TIMEOUT_MS) {
+    const abortTimer = new Timer(readTimeout).start();
+    let message = Buffer.alloc(0);
+    let size = 0;
+
+    do {
+      // eslint-disable-next-line no-await-in-loop, requests to devices are sequential
+      const rawData = await new Promise(resolve => hidDevice.receive(resolve));
+      const packet = Buffer.from(new Uint8Array(rawData));
+
+      // Only process if we get data
+      if (packet.length > 0) {
+        const header = packet.slice(0, 3).toString('ascii');
+        size = packet[3];
+
+        if (header !== BCNLMessage.MAGIC_HEADER) {
+          abortTimer.stop();
+          debug('Invalid packet from Contour device');
+          throw new InvalidMessageError('Unexpected USB packet header.');
+        }
+
+        message = Buffer.concat([message, packet.slice(4)], (message.length + packet.length) - 4);
+      }
+      // USB_BLOCKSIZE - 4, because we don't include the MAGIC_HEADER or the size byte
+    } while (size === (BCNLMessage.USB_BLOCKSIZE - 4));
+
+    abortTimer.stop();
+
+    debug('### READ USB DATA', message.toString('hex'));
+    return message;
```

Comment:
I had the same thought, but when I had less clearTimeout/startTimeout functions lying around, I knew it was worth it. Even more so if we ever have the cause to use `Timer.reset()`